one-page.html

```diff
@@ -1,292 +1,304 @@
+                    <Heading size={2} caps fit textColor="primary" textFont="primary">
+                      Wait what?
+                    </Heading>
+                  </Slide>
+                  <Slide transition={["zoom", "fade"]} bgColor="primary" notes="<ul><li>talk about that</li><li>and that</li></ul>">
+                    <CodePane
+                      lang="jsx"
+                      source={exampleDeck}
+                      margin="20px auto"
+                    />
+                  </Slide>
+                  <Slide>
+                    <ComponentPlayground
+                      theme="dark"
+                    />
+                  </Slide>
+                  <Slide transition={["slide"]} bgImage={"https://cdn.rawgit.com/FormidableLabs/spectacle/master/example/assets/city.jpg"} bgDarken={0.75}>
+                    <Appear fid="1">
+                      <Heading size={1} caps fit textColor="primary">
+                        Full Width
+                      </Heading>
+                    </Appear>
+                    <Appear fid="2">
+                      <Heading size={1} caps fit textColor="tertiary">
+                        Adjustable Darkness
+                      </Heading>
+                    </Appear>
+                    <Appear fid="3">
+                      <Heading size={1} caps fit textColor="primary">
+                        Background Imagery
+                      </Heading>
+                    </Appear>
+                  </Slide>
+                  <Slide transition={["zoom", "fade"]} bgColor="primary">
+                    <Heading caps fit>Flexible Layouts</Heading>
+                    <Layout>
+                      <Fill>
+                        <Heading size={4} caps textColor="secondary" bgColor="white" margin={10}>
+                          Left
+                        </Heading>
+                      </Fill>
+                      <Fill>
+                        <Heading size={4} caps textColor="secondary" bgColor="white" margin={10}>
+                          Right
+                        </Heading>
+                      </Fill>
+                    </Layout>
+                  </Slide>
+                  <Slide transition={["slide"]} bgColor="black">
+                    <BlockQuote>
+                      <Quote>Wonderfully formatted quotes</Quote>
+                      <Cite>Ken Wheeler</Cite>
+                    </BlockQuote>
+                  </Slide>
+                  <Slide transition={["spin", "zoom"]} bgColor="tertiary">
+                    <Heading caps fit size={1} textColor="primary">
+                      Inline Markdown
+                    </Heading>
+                    <Markdown>
+                      {`
+      ![Markdown Logo](https://cdn.rawgit.com/FormidableLabs/spectacle/master/example/assets/markdown.png)
```

Comment:
GLOBAL: The markdown stuff cannot be indented or it won't render. It either has to be deindented like before or use a dedent function something like the following that preserves relative indentation: 

```js
        // Indent to smallest non-empty whitespace level and trim start / end of string.
        const indent = val => {
          let prefix = null;
          return (val || "").split("\n").filter(line => {
            const [cur, remainder] = (line.match(/^([ ]*)([^ ]+)/) || []).slice(1);
            return remainder ? (prefix = null === prefix || cur.length < prefix.length ? cur : prefix,
            !0) : null !== prefix;
          }).map(line => prefix ? line.replace(prefix, "") : line).join("\n").trimRight();
        };
```

Reply:
Thanks for spotting this! Weirdly whitespace-only diffs didn't show up in my vs code version control tab... and then I glossed over the different look of those markdown slides 🤦‍♂ .

Sorry for a potential over-simplification of the ustility function, but would "sanitising" `Markdown` and `MarkdownSlides` with `.replace(/^[ \t]+/gm, '')` not be sufficient? It seems to work for both of these components in this use case.

If so, would you prefer I add this sanitisation to the components themselves, or try to implement a prettier ignore for these sections instead? IMO the former would be more robust

Reply:
For clarity, my usage of that `replace()` is in the two spots contained below:

```js 
<Markdown>
                      {`
      ![Markdown Logo](https://cdn.rawgit.com/FormidableLabs/spectacle/master/example/assets/markdown.png)

      You can write inline images, [Markdown Links](http://commonmark.org), paragraph text and most other markdown syntax
      * Lists too!
      * With ~~strikethrough~~ and _italic_
      * And let's not forget **bold**
                      `.replace(/^[ \t]+/gm, '')}
                    </Markdown>
                  </Slide>
                  {
                    MarkdownSlides(`
      #### Create Multiple Slides in Markdown
      All the same tags and elements supported in <Markdown /> are supported in MarkdownSlides.
      ---
      Slides are separated with **three dashes** and can be used _anywhere_ in the deck. The markdown can either be:
      * A Tagged Template Literal
      * Imported Markdown from another file
                    `.replace(/^[ \t]+/gm, ''))
                  }
```

Reply:
Yep. That's why code snippet has to do the `prefix` inference and uses the heuristic of "first non-whitespace level of indentation is the indentation to strip of the rest".

Put more concisely, we need to transform:

```md
    * one
    * two
        * a
        * b
```

to this:

```md
* one
* two
    * a
    * b
```

not this:

```md
* one
* two
* a
* b
```

I'm not sure if it belongs in spectacle core or just as a snippet in `one-page.html` itself, but I think it's either that or _don't_ autoformat whitespace which in our case creates a correctness problem since the spacing is meaningful.

That snippet I pasted is pretty battle-honed as I use it in nearly all my one page presentations...

Reply:
That said, if we're doing it officially in spectacle (probably don't need for one-page.html because that's up to the user, we should add specific unit tests to cover corner cases like:

- Indentation token to strip occurs again within text and needs to be preserved
- Windows line breaks
- Tabs instead of spaces
- ...etc.

I'm cool with whatever we decide...

Reply:
Saving this discussion in my bookmarks to take into account during re-write! 

For now, have just undone all the prettier indentation changes inside the `<script>`